examples/platform/esp32/common/CHIPDeviceManager.cpp

```diff
@@ -79,6 +79,11 @@ CHIP_ERROR CHIPDeviceManager::Init(CHIPDeviceManagerCallbacks * cb)
     // this function will happen on the CHIP event loop thread, not the app_main thread.
     PlatformMgr().AddEventHandler(CHIPDeviceManager::CommonDeviceEventHandler, reinterpret_cast<intptr_t>(cb));
 
+#if defined(CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING) && CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING
```

Comment:
Shouldn't it just be defined false in the default config (if not already defined), like other config options?  And I see that it is.  Then why do we have the `defined()` check here?

Reply:
I am just following the convention I see all the other checks do. Do we wish to change how we do these checks? If so, maybe that belongs in a new PR.